### ISSUE TITLE
Screen share quality/fps is very low

### ISSUE THREAD

**Issue by vsimeikoKG** [twilio/twilio-video.js#1812]
We have a product built with AWS Chime and we are replacing it with Twilio. So far we had no issues and everything was smooth, but the quality and especially fps of the screenshare is very low when we compare it to what Chime gives us. The high quality screenshare is very important for us and it is currently a deal breaker that stop us from picking the Twilio.

For all our tests we're using 1280x720 screenshare with 30fps. Only one screenshare at the time. We tried small and large group calls and it seems to have no effect on the screenshare quality. We're using YouTube videos with some audio for screenshare tests.

Additionally, we tried to screenshare using Chime and Twilio side by side and Chime is a clear winner. We did these tests with the same amount of participants using the same YouTube video.

**Code to reproduce the issue:**

```js
// We have a Full HD toggle, but you can ignore it here as we always used 720p in our previous tests.
const stream = await navigator.mediaDevices.getDisplayMedia({
  video: {
    cursor: 'always',
    frameRate: {
      ideal: 30,
    },
    width: fullHDScreenShare
      ? {
          ideal: 1920,
          max: 1920,
        }
      : {
          ideal: 1280,
          max: 1280,
        },
    height: fullHDScreenShare
      ? {
          ideal: 1080,
          max: 1080,
        }
      : {
          ideal: 720,
          max: 720,
        },
  },
  audio: {
    latency: 0,
    channelCount: 2,
    echoCancellation: false,
    noiseSuppression: false,
    sampleRate: 48000,
  },
});
```

We tried all possible configuration for the codecs and Bandwidth Profile API:
- VP8 codec with simulcast enabled - screenshare is very choppy for the first ~30 seconds, but it may improve later after switching the simulcast layer.
- VP8 with simulcast and Bandwidth Proflie API - same as above.
- H264 codec with the presentation|collaboration mode - screenshare fps barely goes beyond 10 frames per second.
- H264 with bandwidth profile disabled - seems to be the "best" option so far, but still fps is very unstable.

**Note:** we have a track priority enabled all the time where we give "high" priority to the screenshare, "standard" to the screensharing user, and "low" to everybody else in the call.

Here is what we're using so far as it seems to be give the most "stable" fps for the screenshare:
```js
useEffect(() => {
  if (!twilioAuthToken) {
    return;
  }

  let isMounted = true;
  let twilioRoom: Room;

  connect(twilioAuthToken, {
    // Disable VP8 and simulcast as they produce bad screenshare video quality.
    preferredVideoCodecs: ['H264'],
    // Disable Bandwidth Profile API to avoid low quality screenshare.
    // bandwidthProfile: {
    //   video: {
    //     mode: 'presentation',
    //     trackSwitchOffMode: 'detected',
    //     clientTrackSwitchOffControl: 'auto',
    //     contentPreferencesMode: 'auto',
    //   },
    // },
    networkQuality: {
      local: 1,
      remote: 1,
    },
    tracks: [
      ...(videoMediaStream && store.getState().devices.videoInputEnabled
        ? videoMediaStream.getVideoTracks().map((track) => track.clone())
        : []),
      ...(audioMediaStream && store.getState().devices.audioInputEnabled
        ? audioMediaStream.getAudioTracks().map((track) => track.clone())
        : []),
    ].filter(Boolean),
    logLevel: 'info',
  }).then((room) => {
    if (isMounted) {
      dispatch({type: TwilioHookActions.setRoom, room});

      twilioRoom = room;
    } else {
      disconnectAndCleanupStreams(room);
    }
  });
  return () => {
    isMounted = false;

    disconnectAndCleanupStreams(twilioRoom);
  };
}, [twilioAuthToken]);
```

**Expected behavior:**

Screenshare must have a stable FPS with a good video quality.

**Actual behavior:**

Screenshare FPS mostly stays between 0-15 fps and sometimes may go up to 30fps.

[other_participant.txt](https://github.com/twilio/twilio-video.js/files/8979736/other_participant.txt)
[screensharing_participant.txt](https://github.com/twilio/twilio-video.js/files/8979738/screensharing_participant.txt)

**Software versions:**

- Browser(s): Google Chrome 102.0.5005.115 (Official Build) (x86_64)
- Operating System: MacOS 12.2.1 (21D62)
- twilio-video.js: 2.21.3
- Third-party libraries (e.g., Angular, React, etc.): React 16.14.0

**Devices:**

We're doing our test calls with the group that has high end laptops, such as MacBook Pro 2019-2021 with intel Core i7/M1 Pro CPUs.  However, our end users usually use a bit older devices.

Most of us have a good internet connection with upload/download speed raging from 50mbps to 500mbps.

**Comment by manjeshbhargav** [comment]
Hi @vsimeikoKG ,

Thanks for the report. We are looking into this, and should provide an update soon.

**Comment by manjeshbhargav** [comment]
Hi @vsimeikoKG ,

We have a release candidate that is being tested. I will reach out to you here once I have something to share.

**Comment by manjeshbhargav** [comment]
Hi @vsimeikoKG ,

We just released version 2.22.1 to npm, which has a couple of changes related to screen share tracks. Can you please try it out? Along with this SDK, can you switch your screen capture constraints to the following:
```js
{
  video: true
}
```
and your ConnectOptions to have the following properties:
```js
{
  bandwidthProfile: {
    video: {
      clientTrackSwitchOffControl: 'auto',
      contentPreferencesMode: 'auto'
    }
  },
  preferredVideoCodecs: 'auto'
}
```
What we have observed in our tests is that not specifying capture dimensions and frame rate allows the browser to dynamically manage frame rate based on the content being shared on the screen, and is the optimal behavior for quality.
Please let me know if this works for you.

**Comment by vsimeikoKG** [comment]
Hi @manjeshbhargav,

Thanks for the update.

We recently re-tested screenshare with the fix and it seemed to be way much better!
